File: Parametrizzazione/In/Swap.java
public class Swap { // Passaggio di parametri : in per copia

  public static void swap( int a, int b ) {
    int tmp = 0;

    tmp = a;
    a = b;
    b = tmp;
  }

  public static void main( String[] args ) {

    int a = 5;
    int b = 6;

    System.out.println( "a : " + a );
    System.out.println( "b : " + b );

    swap( a, b );

    System.out.println( "" );
    System.out.println( "a : " + a );
    System.out.println( "b : " + b );

  }
}

File: Parametrizzazione/In/swap.c
#include <stdio.h> // Passaggio di parametri : in per copia

void swap( int a, int b ) {
  int tmp;

  tmp = a;
  a = b;
  b = tmp;
}

int main( void ) {

  int a, b;

  printf( "%s", "a : " );
  scanf( "%d", &a );

  printf( "%s", "b : " );
  scanf( "%d", &b );

  swap( a, b );

  putchar( '\n' );
  printf( "a : %d\n", a );
  printf( "b : %d\n", b );

  return 0;
}
